homeassistant/components/weatherflow/config_flow.py

```diff
@@ -0,0 +1,74 @@
+"""Config flow for WeatherFlow."""
+from __future__ import annotations
+
+import asyncio
+from asyncio import Future
+from asyncio.exceptions import CancelledError
+from typing import Any
+
+from pyweatherflowudp.client import EVENT_DEVICE_DISCOVERED, WeatherFlowListener
+from pyweatherflowudp.errors import AddressInUseError, ListenerError
+
+from homeassistant import config_entries
+from homeassistant.core import callback
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import DOMAIN
+
+
+async def _async_can_discover_devices() -> bool:
+    """Return if there are devices that can be discovered."""
+    future_event: Future[bool] = asyncio.get_running_loop().create_future()
+
+    @callback
+    def _async_found(_):
+        """Handle a discovered device - only need to do this once so."""
+
+        if not future_event.done():
+            future_event.set_result(True)
+
+    async with WeatherFlowListener() as client, asyncio.timeout(10):
+        try:
+            client.on(EVENT_DEVICE_DISCOVERED, _async_found)
+            await future_event
+        except asyncio.TimeoutError:
+            return False
+        except CancelledError:
+            raise ListenerError
+
+    return True
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for WeatherFlow."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle a flow initialized by the user."""
+
+        # Only allow a single instance of integration
+        if self._async_current_entries():
+            return self.async_abort(reason="single_instance_allowed")
+
```

Comment:
I’m not sure what’s intended to be happing here but it doesn’t look like discovery is implemented

Reply:
Spoke to Jeef on Discord, apparently the way you get data from these devices is from UDP broadcasts. My suggestion was to simplify the flow to listening to the broadcasts and if we received the needed info, set up the integration. And if it fails show a message that it failed and submit to try again, what do you think?

Also, I am not experienced with zeroconf so you might also have some ideas for this

Reply:
Is it actually publishing mdns/zeroconf records? You should be able to tell with https://apps.apple.com/de/app/discovery-dns-sd-browser/id305441017

AFAICT the hub is kind enough to broadcasts on port 50222 all of its messages to everyone on the network (even if they don't want them)

Reply:
No discovery, just forced broadcasting

Reply:
As an aside ... I opened a discussion about whether HA should consider UDP based discovery: https://github.com/home-assistant/architecture/discussions/970